agent/engine/docker_task_engine.go

```diff
@@ -464,7 +464,8 @@ func (engine *DockerTaskEngine) sweepTask(task *apitask.Task) {
 }
 
 func (engine *DockerTaskEngine) deleteTask(task *apitask.Task) {
-	for _, resource := range task.Resources {
+	resources := task.GetResources()
+	for _, resource := range resources {
```

Comment:
nit: can this be on a single line `for _, resource := range task.GetResources() { ... `?